libraries/AP_Mount/AP_Mount.cpp

```diff
@@ -63,8 +63,13 @@ void AP_Mount::init()
     // primary is reset to the first instantiated mount
     bool primary_set = false;
 
+#if HAL_MOUNT_STORM32SERIAL_ENABLED || \
+    HAL_MOUNT_SIYI_ENABLED || \
+    HAL_MOUNT_VIEWPRO_ENABLED || \
+    HAL_MOUNT_TOPOTEK_ENABLED
     // keep track of number of serial instances for initialisation
     uint8_t serial_instance = 0;
+#endif
```

Comment:
There's an alternative to the #ifs here, and that's:
```suggestion
        (void)serial_instance;
#endif
```

It means when someone adds another serial driver they don't have to add their new thing here.

Thoughts?